ccm_web/client/src/utils/GradebookProcessor.tsx

```diff
@@ -0,0 +1,141 @@
+import { CSVRecord } from '../utils/FileParserWrapper'
+import { InvalidationType } from '../models/models'
+
+export interface GradebookUploadRecord extends CSVRecord {
+  'SIS Login ID': string
+}
+
+export const REQUIRED_LOGIN_ID_HEADER = 'SIS Login ID'
+const OTHER_REQUIRED_HEADERS = ['Student Name', 'Student ID', 'SIS User ID', 'Section']
+export const REQUIRED_ORDERED_HEADERS = [
+  ...OTHER_REQUIRED_HEADERS.slice(0, 3), REQUIRED_LOGIN_ID_HEADER, OTHER_REQUIRED_HEADERS[3]
+]
+
+export const isGradebookUploadRecord = (record: CSVRecord): record is GradebookUploadRecord => {
+  return typeof record['SIS Login ID'] === 'string'
+}
+
+export interface GradebookInvalidation {
+  message: string
+  type: InvalidationType
+}
+
+interface ProcessResultSuccess {
+  valid: true
+  processedRecords: GradebookUploadRecord[]
+  assignmentHeader: string
+  invalidations: GradebookInvalidation[]
+}
+
+interface ProcessResultFailure {
+  valid: false
+  invalidations: GradebookInvalidation[]
+}
+
+export type ProcessResult = ProcessResultSuccess | ProcessResultFailure
+
+export default class GradebookProcessor {
```

Comment:
Currently this is used by 3rd-party GB, and I don't think most of the code used here will be shared with Canvas GB converter, this could be named as `ThirdPartyGradebookProcessor`. This is for code reading clarity purpose since we have 2 gradebook tools.

Reply:
I can change that. I wasn't sure whether someday this class could be used in both.

Reply:
Done, see https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/223/commits/ae5c9174a772f09eee36a9e685919795342bf7bd

Reply:
Thanks!